### ISSUE TITLE
关于学习率设置的问题

### ISSUE THREAD

**Issue by ThomaswellY** [sssaury/HAM#2]
您好作者！论文中您做的预训练模型在reid下游任务上的finetune效果特别好，但我有一点困惑。
您的readme文件中说“Initialize the model weights with our ReID pretrained model and reduce the learning rate by half”
而您的论文中说“The ReID model is trained using the Adam optimizer with a learning rate of 1e-5”，请问如果要复现您的下游finetune结果，是不是只需要调整IRRA中的epoch（设为30）、batchsize(设为每张卡128)和学习率呢（设为5e-6还是沿用默认的1e-5呢？）

**Comment by ThomaswellY** [comment]
这是我的实验结果，有错误的地方还请您指点一下

**Comment by sssaury** [comment]
你好，我们在预训练的时候是八卡每张128bs，lr为1e-5训练，下游finetune时是用的单卡128bs、60epoch、lr5e-6训练

**Comment by ThomaswellY** [comment]
> 你好，我们在预训练的时候是八卡每张128bs，lr为1e-5训练，下游finetune时是用的单卡128bs、60epoch、lr5e-6训练

作者您好，我发现您提供的预训练模型是module结构的，所以说您是DDP预训练好的吧？那么您在下游做finetune的时候，为什么还是用单卡跑呢？

**Comment by sssaury** [comment]
epoch不够大可能是一个原因，另外我们当时做完预训练之后因为有很多下游finetune实验包括调参要同时进行，所以才选择用单卡，后面确实发现多卡finetune性能会比单卡稍弱一些

**Comment by ThomaswellY** [comment]
@sssaury 谢谢你的回复，如果可以的话，我想了解下CUHK的具体超参设置。ICFG和RSTP我已经顺利复现了您论文中的效果，这点十分感谢！但是CUHK我即使设置了多个初始seed，最好的R1也是77.2，我想是不是有什么地方遗漏了，所以没有达到您论文总的77.7的结果。如果您能提供更详细的信息就更好了！

**Comment by NickyyChen** [comment]
> ![Image](https://github.com/user-attachments/assets/ec684f3f-8b98-462d-834a-ada385c301d0)
> 
> 这是我的实验结果，有错误的地方还请您指点一下

你好，你的下游finetune代码是怎么样的，我发现训练没达到预期你的效果，不知道哪里出错了

**Comment by sssaury** [comment]
finetune代码就是irra的代码&nbsp; bs=128 1gpu lr=5e-6 epoch=60 ，finetune的log和模型会整理开源在github，里面有详细的超参设置可以参考


JiangJiayu
***@***.***



        



         原始邮件
         
       
发件人：nickychen ***@***.***&gt;
发件时间：2025年5月8日 09:27
收件人：sssaury/HAM ***@***.***&gt;
抄送：sssaury ***@***.***&gt;, Mention ***@***.***&gt;
主题：Re: [sssaury/HAM] 关于学习率设置的问题 (Issue #2)



NickyyChen left a comment (sssaury/HAM#2)

image.png (view on web)

这是我的实验结果，有错误的地方还请您指点一下

你好，你的下游finetune代码是怎么样的，我发现训练没达到预期你的效果，不知道哪里出错了

**Comment by NickyyChen** [comment]
![Image](https://github.com/user-attachments/assets/01939923-2e80-4829-80b4-058728ccab30)

**Comment by NickyyChen** [comment]
> ![Image](https://github.com/user-attachments/assets/ec684f3f-8b98-462d-834a-ada385c301d0)
> 
> 这是我的实验结果，有错误的地方还请您指点一下


> finetune代码就是irra的代码&nbsp; bs=128 1gpu lr=5e-6 epoch=60 ，finetune的log和模型会整理开源在github，里面有详细的超参设置可以参考
> 
> 
> JiangJiayu
> ***@***.***
> 
> 
> 
> 
> 
> 
> 
>          原始邮件
> 
> 
> 发件人：nickychen ***@***.***&gt;
> 发件时间：2025年5月8日 09:27
> 收件人：sssaury/HAM ***@***.***&gt;
> 抄送：sssaury ***@***.***&gt;, Mention ***@***.***&gt;
> 主题：Re: [sssaury/HAM] 关于学习率设置的问题 (Issue [#2](https://github.com/sssaury/HAM/issues/2))
> 
> 
> 
> NickyyChen left a comment ([sssaury/HAM#2](https://github.com/sssaury/HAM/issues/2))
> 
> image.png (view on web)
> 
> 这是我的实验结果，有错误的地方还请您指点一下
> 
> 你好，你的下游finetune代码是怎么样的，我发现训练没达到预期你的效果，不知道哪里出错了
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.

![Image](https://github.com/user-attachments/assets/630c592b-26f3-4ba4-a5d2-cde191255d5b)

![Image](https://github.com/user-attachments/assets/255e1cfa-15ee-4531-aec7-e4dfa027a031)

第一张图是我用作者前天新发的best0复现的结果，基本跟论文符合了，不过之前发的best0-2（第二张图）在CUHK数据集上没有这个新的best0的效果好，不知道是怎么回事，值得注意的是irra原论文milestone是20-50，这里作者改到20-40。我好奇的是，作者新发的best0内存大小1.35GB，原本的best0-2大小是1002.67MB，不知道这个新发的best0是训练的时候有什么改动吗？

**Comment by sssaury** [comment]
random100w_2HAMcaption（原本开源的best0-2）是100w HAM-PEDES数据预训练clip的权重，后面开源的ham_finetuneIRRAcuhk5e-6_bs128是用前面best0作为预训练权重finetune IRRA+cuhk后的参数，比clip多了IRR模块的参数。
